cli/cmd/asset_create.go

```diff
@@ -0,0 +1,148 @@
+		if err != nil {
+			logger.Fatalf("Unable to get update-if-exists flag vaule: %v", err)
+		}
+		jsonPath, err := cmd.Flags().GetString("jsonpath")
+		if err != nil {
+			logger.Fatalf("Unable to get jsonpath: %v", err)
+		}
+
+		_, err = assetType.ValueByDiscriminator()
+		if err != nil {
+			logger.Fatalf("Failed to determine asset type: %v", err)
+		}
+
+		asset, err := createAsset(context.TODO(), assetType, server, updateIfExist)
+		if err != nil {
+			logger.Fatalf("Failed to create asset: %v", err)
+		}
+
+		if err := cliutils.PrintJSONData(asset, jsonPath); err != nil {
+			logger.Fatalf("Failed to print jsonpath: %v", err)
+		}
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(assetCreateCmd)
+
+	assetCreateCmd.Flags().String("from-json-file", "", "asset json filename")
+	assetCreateCmd.Flags().String("server", "", "VMClarity server to create asset to, for example: http://localhost:9999/api")
+	assetCreateCmd.Flags().Bool("update-if-exists", false, "the asset will be updated the asset if it exists")
+	assetCreateCmd.Flags().String("jsonpath", "", "print selected value of asset")
+	if err := assetCreateCmd.MarkFlagRequired("from-json-file"); err != nil {
+		logger.Fatalf("Failed to mark from-json-file flag as required: %v", err)
+	}
+	if err := assetCreateCmd.MarkFlagRequired("server"); err != nil {
+		logger.Fatalf("Failed to mark server flag as required: %v", err)
+	}
+}
+
+func getAssetFromJSONFile(filename string) (*models.AssetType, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %v", err)
+	}
+	defer file.Close()
+
+	// get the file size
+	stat, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file stat: %v", err)
+	}
+	// read the file
+	bs := make([]byte, stat.Size())
+	_, err = file.Read(bs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read file: %v", err)
+	}
+
+	assetType := &models.AssetType{}
+	if err := assetType.UnmarshalJSON(bs); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal asset into AssetType %v", err)
+	}
+
+	return assetType, nil
+}
+
+func createAsset(ctx context.Context, assetType *models.AssetType, server string, updateIfExist bool) (*models.Asset, error) {
+	client, err := backendclient.Create(server)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create VMClarity API client: %w", err)
+	}
+
+	creationTime := time.Now()
+	assetData := models.Asset{
+		AssetInfo: assetType,
+		LastSeen:  &creationTime,
+		FirstSeen: &creationTime,
+	}
+	asset, err := client.PostAsset(ctx, assetData)
+	if err == nil {
+		return asset, nil
+	}
+	var conflictError backendclient.AssetConflictError
+	// As we got a conflict it means there is an existing asset
+	// which matches the unique properties of this asset, in this
+	// case if the update-if-exists flag is set we'll patch the just AssetInfo and FirstSeen instead.
+	if !(errors.As(err, &conflictError) && updateIfExist) {
```

Comment:
nit: This reads a little bit better when trying understand the code. "If its not a conflict error or we're not updating if exists then error"
if !errors.As(err, &conflictError) || !updateIfExists {
}